### PR TITLE
TECH-574 added converted project to typescript

### PR DESCRIPTION
**Pull request by ikhoaA**

### CODE REVIEW

File: lib/ImageGallery/ImageGallery.tsx [hunk 1]
```diff
@@ -76,7 +100,7 @@ const ImageGallery = ({
       }
     });
 
-  const handleThumbnailChange = (index) => {
+  const handleThumbnailChange = (index: any) => {
```
Comment (loan-laux) [inline lib/ImageGallery/ImageGallery.tsx:103]: Isn't the `index` a `number`?

File: lib/utils/types.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  isPrimary: boolean
+  imageUrl: string
+  alt: string
+  Image: React.FC
+}
+
+export interface PricingClassNames {
+  promotional: string
+  original: string
+}
```
Comment (loan-laux) [inline lib/utils/types.ts:20]: Make sure to leave a blank line at the end of every file as best practice to comply with [POSIX standards](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206)

**Review by loan-laux — CHANGES_REQUESTED** [review]
Looking pretty good @ikhoaA! A few requests for you — most of which will be solved by running Prettier on this branch. Also, please make sure to commit the new `package-lock.json`. Could you also explain why we're building for both CJS and ESM? Just curious about this approach as I'm not an expert on TS. Thanks!

File: package.json [hunk 3]
```diff
@@ -11,9 +15,13 @@
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook",
     "next-dev": "next dev",
+    "build": "rm -rf /dist && npm run build:esm && npm run build:cjs && npm run next-build",
+    "build:esm": "node_modules/typescript/bin//tsc --preserveWatchOutput --jsx react-jsx",
```
[the remark this answers is not in the document]
Comment (ikhoaA) [reply package.json:19]: resolved!

File: lib/Button/Button.tsx [hunk 4]
```diff
@@ -2,12 +2,23 @@ import React from 'react';
...
-const Button = ({ classNames, disabled, displayText, onClick }) => (
+interface ClassNames {
+  button?: string
+}
+interface Props {
+  classNames?: ClassNames,
+  disabled: boolean,
+  displayText: string,
+  onClick: () => {},
+  OverrideClasses?: boolean
```
[the remark this answers is not in the document]
Comment (ikhoaA) [reply lib/Button/Button.tsx:13]: Resolved

**Comment by ikhoaA** [comment]
> 

Hi @loan-laux , the reason for building both CJS and ESM versions of the code is our package to be easily used for vanilla JavaScript and TypeScript.

**Review by loan-laux — CHANGES_REQUESTED** [review]
All good, thanks! Only one request: please rename `OverrideClasses` to `overrideClasses` throughout this branch. I'll instantly merge when done. Great work @ikhoaA!

**Review by loan-laux — APPROVED** [review]
Looking good! Thanks @ikhoaA

**Pull request merged**